src/WebJobs.Script/Rpc/LanguageWorkerProcess.cs

```diff
@@ -184,8 +190,9 @@ internal void HandleWorkerProcessExitError(LanguageWorkerProcessExitException la
             // The subscriber of WorkerErrorEvent is expected to Dispose() the errored channel
             if (langExc != null && langExc.ExitCode != -1)
             {
-                _workerProcessLogger.LogDebug(langExc, $"Language Worker Process exited.", _process.StartInfo.FileName);
-                _eventManager.Publish(new WorkerErrorEvent(_runtime, _workerId, langExc));
+                _workerProcessLogger.LogDebug(langExc, $"Language Worker Process exited (exit code: {langExc.ExitCode}).", _process.StartInfo.FileName);
+                var isIntentionalExit = langExc.ExitCode == NonErrorExitCode;
+                _eventManager.Publish(new WorkerErrorEvent(_runtime, _workerId, isIntentionalExit ? null : langExc));
```

Comment:
instead of not passing the exception, let function dispatcher decide weather to ignore the erorr or not by checking the exit code.

Reply:
@pragnagopa Thank you, Pragna. What I like about this idea is that it will make my change simple and local. However, this will also introduce a dependency of `FunctionDispatcher` on the worker process: the fact that there is such a thing as a worker process, and the fact that its exit codes may have a special meaning. Not necessarily a big deal here, but it feels like these details belong to `LanguageWorkerProcess` instead, and they should be communicated to `FunctionDispatcher` in a more high-level form, saying something like "this WorkerError does not actually indicate a problem", and this is why I structured the code this way. At least in theory, `FunctionDispatcher` could handle in-process workers (and they also may have internal reasons to be restarted), and this dependency would be awkward...

I can still do as you suggest, just want to make sure you are aware of this consideration.

Reply:
Thanks for the explanation. I consider LanguageWorkerProcess to be a light weight wrapper for process related event handlers. FunctionDispatcher should have logic and control over decisions on when to restart a process. Let me know if you would like to discuss this further.

Reply:
Implemented a new event instead.